MVP Module/Tests/Feed UI/FeedUIIntegrationTests.swift

```diff
@@ -280,6 +280,43 @@ final class FeedUIIntegrationTests: XCTestCase {
 		wait(for: [exp], timeout: 1.0)
 	}
 
+	func test_loadFeedWithError_showsErrorOnCompletion() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+
+		XCTAssertNil(sut.errorMessage)
+
+		loader.completeFeedLoadingWithError()
+		XCTAssertEqual(sut.errorMessage, localized("FEED_VIEW_CONNECTION_ERROR"))
+	}
+
+	func test_hidesErrorView_onTappingErrorView() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+
+		XCTAssertNil(sut.errorMessage)
+
+		loader.completeFeedLoadingWithError()
+
+		sut.simulateTapOnErrorMessage()
```

Comment:
There's no value in checking if `sut.errorMessage` is nil before completion with an error. It is already checked by `test_loadFeedWithError_showsErrorOnCompletion` thus `XCTAssertNil(sut.errorMessage)` can be removed.

At the same time the test checks that the error is hidden after some action, which in turn means that it should not be hidden before the action. So as a suggestion, check for non-empty error message can be added right before `simulateTapOnErrorMessage` only to make the test more clear. 

```suggestion
		loader.completeFeedLoadingWithError()

		XCTAssertEqual(sut.errorMessage, localized("FEED_VIEW_CONNECTION_ERROR"))
		sut.simulateTapOnErrorMessage()
```

Reply:
done